### PR TITLE
C36+C50 model confirmation + color bug fixed

### PR DESCRIPTION
**Pull request by Alexkszzz**
## What?
I've added the undo button and model confirmation to treat the cases when users accidentally remove or add students to inSession table. For more background, see #36. I also added Jordan's fix to color problem #50.

## How?
There are 2 main parts in this change, an Undo button to move students from inSession back to the waiting queue, and a confirmation box for when removing students from the waiting queue.

## Testing 
Tested manually myself, works for both backend and frontend.

### CODE REVIEW

File: app/app/static/custom/ourscript.js [hunk 1]
```diff
@@ -211,13 +227,13 @@ function setTime(id) {
...
-	if (timers[id] == 600) {
+	if (timers[id] >= 600) {
 		document.getElementById(id).className = 'yellowTime';
 	}
-	if (timers[id] == 1200) {
+	if (timers[id] >= 1200) {
 		document.getElementById(id).className = 'orangeTime';
 	}
-	if (timers[id] == 1800) {
+	if (timers[id] >= 1800) {
```
Comment (frinzekt) [inline app/app/static/custom/ourscript.js:236]: this should be of different pull request. A pull request should be tied to the change it was addressing

File: app/app/queue.py [hunk 2]
```diff
@@ -45,7 +43,7 @@ def update_entry(entry_id):
 
     if src == 'Ended':
         if status == 'In Queue':
-            entry.exitSessionTime = Null
+            entry.exitSessionTime = None
```
Comment (frinzekt) [inline app/app/queue.py:46]: looks good

File: app/app/static/custom/ourscript.js [hunk 3]
```diff
@@ -50,15 +51,29 @@ const moveToSessionList = async(data) => {
...
+			const index = queueList['In Session'].findIndex((element) => element.id == dataResponse.id)
+			// pop the data from in Session queue
+			queueList['In Session'].splice(index, 1)
+			queueList[dataResponse.queue].push(dataResponse)
+			clearInterval(timerIntervals[dataResponse.id]);
+			const timeDifferenceMiliseconds = (new Date()).getTime() - (new Date(dataResponse.enterQueueTime)).getTime()
+			timers[dataResponse.id] = Math.round(timeDifferenceMiliseconds / 1000);
+			timerIntervals[dataResponse.id] = setInterval(setTime, 1000, dataResponse.id);
+		}
+
```
Comment (frinzekt) [inline app/app/static/custom/ourscript.js:76]: looks good!

File: app/app/static/custom/ourscript.js (hunk 1 above)
Comment (Alexkszzz) [reply app/app/static/custom/ourscript.js:236]: yeah, I was thinking of combining it since it's just a small change. Probably should leave it for that branch for a cleaner GitHub view

Comment (frinzekt) [reply app/app/static/custom/ourscript.js:236]: Your choice... Your pr is good enough... If you want to split it it's your choice

**Review by frinzekt — APPROVED** [review]

File: app/app/static/custom/ourscript.js (hunk 1 above)
Comment (frinzekt) [reply app/app/static/custom/ourscript.js:236]: Merge whenever you are ready

Comment (Alexkszzz) [reply app/app/static/custom/ourscript.js:236]: I'll note C50 in the PR name as well, should be good.
